### PR TITLE
[bitnami/rabbitmq] Fix NetworkPolicy to include extraContainerPorts in ingress rule

### LINKED ISSUE #25426: [bitnami/rabbitmq] Not able to connect to web-stomp web socket on port 15674
### Name and Version

bitnami/rabbitmq:14.0.2

### What architecture are you using?

amd64

### What steps will reproduce the bug?

I followed the suggestions provided in this thread 

> https://github.com/bitnami/charts/issues/6427

in order to enable web-stomp plugin, add extra container ports, add extra service ports.
Everything is ok because If I do a port-forward from my laptop to the service on port 15674, I can connect to the web socket like a charm using the url: **127.0.0.1:15674/ws/**

```
kubectl port-forward --namespace rabbitmq svc/opbg-rabbitmq 15674:15674 --kubeconfig=/Users/alexsam/mywork/opbg/opbg-sviluppo.yaml
Forwarding from 127.0.0.1:15674 -> 15674
Forwarding from [::1]:15674 -> 15674
Handling connection for 15674
```

To make the rabbitmq web socket available to clients outside of the cluster, I created an ingress in which I simply proxy the requests on the path /ws/ to the service on web-stomp port 15674.
```
spec:
  rules:
    - host: rabbitmq-dev.subdomain.it
      http:
        paths:
          - backend:
              service:
                name: opbg-rabbitmq
                port:
                  number: 15674
            path: /ws/
            pathType: Prefix
```

But I am not able to connect to the web socket anymore. The url I am trying is: **rabbitmq-dev.subdomain.it/ws/**

Consider that the ootb ingress created by helm, the one that I can use to access the management rabbitmq console, works fine.

I even used the suggestion at the bottom of the same thread 6427, and added an extra rule property to the helm values of the chart. In this case, the ingress created by Helm was:

```
spec:
  rules:
    - host: rabbitmq-dev.subdomain.it
      http:
        paths:
          - backend:
              service:
                name: opbg-rabbitmq
                port:
                  name: http-stats
            path: /
            pathType: ImplementationSpecific
    - host: stomp-rabbitmq-dev.subdomain.it
      http:
        paths:
          - backend:
              service:
                name: opbg-rabbitmq
                port:
                  name: stomp-ws
            path: /ws/
            pathType: ImplementationSpecific
```

But it's not working. In this case, I even created another entry in the DNS in order to have two different hostnames, resolved with the same IP.
At this point, I was thinking that there was something that it's not working or maybe I was missing something in the Ingress configuration.
So I made the last try, creating a service of type NodePort:
```
  ports:
    - name: webstomp-np
      nodePort: 30674
      port: 15674
      protocol: TCP
      targetPort: 15674
```
But it doesn't work like that either, trying to connect to the node on which the rabbitmq pod has been scheduled to start.

I am struggling with this from days.
Could you please tell me something more on this?

Thanks

### Are you using any custom parameters or values?

1. added extra plugin 

`extraPlugins: rabbitmq_auth_backend_ldap rabbitmq_web_stomp`

2. added extra container port:

```
extraContainerPorts:
      - containerPort: 15674
        name: stomp-ws
```

3. added  extra port to the service

```
    service:
      extraPorts:
        - name: stomp-ws
          port: 15674
          protocol: TCP
          targetPort: stomp-ws
```

4. even added extra rules to the ingress

```
  extraRules:
    - host: stomp-rabbitmq-dev.subdomain.it
      http:
        paths:
          - backend:
              service:
                name: opbg-rabbitmq
                port:
                  name: stomp-ws
            path: /ws/
            pathType: ImplementationSpecific
```

### What is the expected behavior?

To be able to connect to web socket exposed by RabbitMQ web-stomp plugin on port 15674, using an ingress.

Just to be precise, my K8s cluster is based on RKE2 distribution (k8s version is v1.27.10 +rke2r1).

### What do you see instead?

When I try to connect to the web stomp service, I obtain a 504 Gateway timeout, in my opinion the ingress is not able to communicate with the service on that port.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ChrisRitchie**
Fix to extraContainerPorts not being pulled in from values file

Signed-off-by: Chris Ritchie <chrisritchie@zoho.com>

### Description of the change

Currently any extraContainerPorts configured within the values.yaml file are ignored and fail to be added to the networkingpolicy.yaml file. 

For example, without this fix the following config in the values.yaml file is ignored:
```
extraContainerPorts:
- name: stomp-ws
  containerPort: 15674
- name: stomp-tcp
  containerPort: 61613
```

### Benefits

stomp-tcp & stomp-ws ports can be exposed using an ingress. 

### Applicable issues

- fixes #25426

### Additional information

This did work at some point in the past (~2021), and the `extraContainerPorts` config in the values.yaml file still exists. As this used to work at some point in the past I consider this a bug rather than an enhancement.

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/rabbitmq/templates/networkpolicy.yaml [hunk 1]
```diff
@@ -59,6 +59,9 @@ spec:
         - port: {{ .Values.containerPorts.amqpTls }}
         - port: {{ .Values.containerPorts.dist }}
         - port: {{ .Values.containerPorts.manager }}
+        {{ - range .Values.extraContainerPorts }}
+        - port: {{ . }}
+        {{ - end }}
```
Comment (migruiz4) [inline bitnami/rabbitmq/templates/networkpolicy.yaml:64]: This template is invalid, it should use `{{-` not `{{ -`
```suggestion
        {{- range .Values.extraContainerPorts }}
        - port: {{ . }}
        {{- end }}
```

**Review by migruiz4 — COMMENTED** [review]
Hi @ChrisRitchie,

Thank you for your contribution! Could you please address the issue I mentioned in my comment?

File: bitnami/rabbitmq/templates/networkpolicy.yaml (hunk 1 above)
Comment (ChrisRitchie) [reply bitnami/rabbitmq/templates/networkpolicy.yaml]: OK thanks, sorry about that. I have pushed the updated template formatting.

**Comment by ChrisRitchie** [comment]
Thanks @migruiz4 for your review. I have updated the PR as requested. Is there anything else I need to do?

File: bitnami/rabbitmq/templates/networkpolicy.yaml [hunk 2]
```diff
@@ -59,6 +59,9 @@ spec:
         - port: {{ .Values.containerPorts.amqpTls }}
         - port: {{ .Values.containerPorts.dist }}
         - port: {{ .Values.containerPorts.manager }}
+        {{- range .Values.extraContainerPorts }}
+        - port: {{ . }}
```
Comment (migruiz4) [inline bitnami/rabbitmq/templates/networkpolicy.yaml:63]: I have noticed that the format of the extraContainerPorts is
```yaml
extraContainerPorts:
- name: new_port_name
  containerPort: 1234
```

Therefore using `{{ . }}` would result in invalid syntax after rendering.

```suggestion
        - port: {{ .containerPort }}
```

**Comment by ChrisRitchie** [comment]
Hi @migruiz4, code review changes implemented and conflicts resolved. Thanks

**Pull request closed without merging**